### ISSUE TITLE
Feature tests

### ISSUE THREAD

**Issue by cdmo** [psu-libraries/psulib_blacklight#233]
Create feature tests for:
- [x] Title, author, subject dropdown searches
- [x] Advanced search
- [x] _search_results.html.erb 
- [x] Availability - handled with #450 
- [x] Subjects
- [x] Genre
- [x] Publication fields

Create tests for our behaviors (javascript) to make sure things don't break when we update javascript code or  JS dependencies (like Bootstrap)

**Comment by banukutlu** [comment]
Make sure to cover https://github.com/psu-libraries/psulib_blacklight/wiki/Regression-Testing and https://github.com/psu-libraries/psulib_blacklight/wiki/Acceptance-Tests

**Comment by cdmo** [comment]
> Create some tests for our behaviors (javascript) to make sure things don't break when we update javascript code or js dependencies (like Bootstrap)

Well, I did this part of the issue for book covers using `jest` and importing fixtured data during CI. It tests if the JavaScript is functioning at a unit level (jest) and then makes sure the actual image from Google Books shows up on the page by using a selenium javascript-enabled chromedriver test. I made this a feature test even though there aren't any clicks or interactions. Think it's okay because the implied integration is the JS working behind the scenes to do the JSONP "AJAX" work and replacement of the tag. So that's one thing anyway.

Wonder if we should make break apart the Regression Testing and Acceptance Tests wiki pages into new issues. Also, we should talk about using Cucumber versus just representing acceptance tests in rspec capybara feature tests. Do we want the cucumber overhead, I dont know.
